core/types/transaction.go

```diff
@@ -171,7 +163,7 @@ func (tx *Transaction) CheckAccountPermission() error {
 		to = *tx.To()
 	}
 
-	return IsTransactionAllowed(tx.From(), to, tx.Value(), tx.GasPrice(), big.NewInt(int64(tx.Gas())), tx.Data(), transactionType)
+	return cache.IsTransactionAllowed(tx.From(), to, tx.Value(), tx.GasPrice(), big.NewInt(int64(tx.Gas())), tx.Data(), transactionType)
```

Comment:
This is somehow misleading because one can think that the check is performed on cache while it is performed on the contract.

Maybe a better option would be to move `IsTransactionAllowed` and `PermissionTransactionAllowedFunc` under package `permission`

Reply:
agree. refactored a bit and renamed `cache` to `core` and moved `types` under `core`. The package structure looks cleaner now

Reply:
done